### PR TITLE
SWC-6536: use larger runner for e2e tests

### PR DESCRIPTION
**Pull request by hallieswan**

### CODE REVIEW

File: .github/workflows/build-test-e2e.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on: push
 jobs:
   build-and-run-e2e:
     if: ${{ github.ref_name == 'develop' || github.ref_name == 'release-**' || github.actor == github.repository_owner }}
-    runs-on: macos-latest
+    runs-on: ${{ github.repository_owner == 'Sage-Bionetworks' && 'ubuntu-22.04-4core-16GBRAM-150GBSSD' || 'macos-latest' }}
```
Comment (hallieswan) [inline .github/workflows/build-test-e2e.yml:7]: Use larger runner when e2e tests are run in Sage-Bionetworks organization, but continue using `macos` runner in dev forks

Comment (nickgros) [reply .github/workflows/build-test-e2e.yml:7]: Good idea. Let's do this in the monorepo as well

File: .github/workflows/build-test-e2e.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ on: push
 jobs:
   build-and-run-e2e:
```
Comment (nickgros) [inline .github/workflows/build-test-e2e.yml:5]: I think you had a timeout at the application level, so maybe it's not necessary, but do you think one should be added at the job level?

**Review by nickgros — APPROVED** [review]

File: .github/workflows/build-test-e2e.yml (hunk 2 above)
Comment (hallieswan) [reply .github/workflows/build-test-e2e.yml:5]: Oh good call! I'll add that now

File: .github/workflows/build-test-e2e.yml (hunk 1 above)
Comment (nickgros) [reply .github/workflows/build-test-e2e.yml:7]: https://sagebionetworks.jira.com/browse/PORTALS-2798

**Pull request merged**